lxd/auth/encryption/sign.go

```diff
@@ -0,0 +1,58 @@
+package encryption
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/golang-jwt/jwt/v5"
+	"github.com/google/uuid"
+
+	"github.com/canonical/lxd/shared"
+)
+
+const (
+	// defaultExpiry is used to set an expiry of "never", meaning 10 years.
+	defaultExpiry = "10y"
+)
+
+// GetSignedJWT generates and signs a token. If a UUID salt is provided, a unique signing key is derived from the secret using
```

Comment:
Please can you expand this docblock to explain the different use-cases for the different modes this function provides (salted and unsalted) - perhaps even it should be split into different functions, one for each purpose.

Right now the docblock is a bit abstract and depends on the caller understanding the implications of the different modes, which I feel we could improve here and provide guidance around the use-cases.

Reply:
I've simplified this to only issue DevLXD tokens and added a comment about what it creates including how the claims are populated.